File: ios/Runner/CalculatorApiImpl.swift
import Foundation

class CalculatorApiImpl: CalculatorApi {
    func add(a: Double, b: Double) throws -> Double {
        return a + b
    }
    
    func subtract(a: Double, b: Double) throws -> Double {
        return a - b
    }
    
    func multiply(a: Double, b: Double) throws -> Double {
        return a * b
    }
    
    func divide(a: Double, b: Double) throws -> Double {
        if b == 0 {
            throw PigeonError(code: "DIVISION_BY_ZERO", message: "Cannot divide by zero", details: nil)
        }
        return a / b
    }
}

File: android/app/src/main/kotlin/com/example/calculator_native/CalculatorApiImpl.kt
package com.example.calculator_native

class CalculatorApiImpl : CalculatorApi {
    override fun add(a: Double, b: Double): Double {
        return a + b
    }

    override fun subtract(a: Double, b: Double): Double {
        return a - b
    }

    override fun multiply(a: Double, b: Double): Double {
        return a * b
    }

    override fun divide(a: Double, b: Double): Double {
        if (b == 0.0) {
            throw FlutterError("DIVISION_BY_ZERO", "Cannot divide by zero")
        }
        return a / b
    }
}
